pygmsh/built_in/geometry.py

```diff
@@ -110,6 +110,39 @@ def add_line_loop(self, *args, **kwargs):
         self._GMSH_CODE.append(p.code)
         return p
 
+    def add_unordered_line_loop(self, lines):
+        # Categorise graph edges by their vertex pair ids
+        point_pair_ids = numpy.array(
+            [[line.points[0].id, line.points[1].id] for line in lines]
+        )
+
+        # Indices of reordering
+        order = numpy.arange(len(point_pair_ids), dtype=int)
+        # Compute orientations where orient[j] == -1 requires edge j to be reversed
+        orient = numpy.array([1] * len(point_pair_ids), dtype=numpy.int)
```

Comment:
This better be a Boolean array, `is_oriented` or so. Instead of multiplying by `-1` below, you can simply negate (`~` or `logical_not`).